crowbar_framework/lib/crowbar/installer.rb

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Installer
+    class << self
+      def install
+        crowbar_dir = Rails.root.join("..")
+        case NodeObject.admin_node[:platform_family]
+        when "suse"
+          cmd = "sudo #{crowbar_dir}/bin/install-chef-suse.sh --crowbar"
+        else
+          return [501, I18n.t(".system_not_supported", scope: "installer.index")]
+        end
+
+        # spawn the script asynchronously in the background
+        pid = spawn(cmd)
+        Process.detach(pid)
+
+        return [200, ""]
+      end
+
+      def all_steps
+        [
+          :pre_sanity_checks,
+          :run_services,
+          :initial_chef_client,
+          :barclamp_install,
+          :bootstrap_crowbar_setup,
+          :apply_crowbar_config,
+          :transition_crowbar,
+          :chef_client_daemon,
+          :post_sanity_checks
+        ]
+      end
+
+      def crowbar_lib_dir
```

Comment:
s/crowbar_lib_dir/lib_path/ or s/crowbar_lib_dir/lib_dir/. I would prefer _path as it's a pathname.